runtime-decompiler/src/main/java/org/jrd/frontend/utility/TeeOutputStream.java

```diff
@@ -0,0 +1,54 @@
+package org.jrd.frontend.utility;
+
+
+import java.io.ByteArrayOutputStream;
+import java.io.PrintStream;
+import java.nio.charset.StandardCharsets;
+
+
+/**
+ * Behaves like the 'tee' command, sends output to both actual std stream and a
+ * log
+ */
+public final class TeeOutputStream extends PrintStream {
+
+    // Everything written to TeeOutputStream is written to this buffer too
+    private final ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream();
+
+    public TeeOutputStream(PrintStream stdStream) {
+        super(stdStream, false, StandardCharsets.UTF_8);
+    }
+
+    @Override
+    public void close() {
+        super.close();
+    }
+
+    @Override
+    public void flush() {
+        super.flush();
+    }
```

Comment:
These overridden methods don't do anything - by default super.method() is called when a subclass doesn't override the method. You can remove them, add an explanatory comment of why they are there (clarity?) or add some code into them that would justify their existence (closing the custom stream?)

Reply:
Yup. i was originaly thinking about closing/flusing bytearray output stream but realised that that one  do nothing in close/flush. Will remove